src/passes/Precompute.cpp

```diff
@@ -174,6 +174,38 @@ struct Precompute
     } while (propagate && worked);
   }
 
+  template<typename T> void reuseConstantNode(T* curr, Flow flow) {
+    if (flow.values.isConcrete()) {
+      // reuse a const / ref.null / ref.func node if there is one
+      if (curr->value && flow.values.size() == 1) {
+        Literal singleValue = flow.getSingleValue();
+        if (curr->value->template is<Const>() && singleValue.type.isNumber()) {
```

Comment:
Can do
```
if (singleValue.isNumber()) {
  if (auto* c = curr->value->template dynCast<Const>()) {
    ..
  }
}
```
(to avoid both an `is` and a `cast`) and also later down.

Reply:
(and then do an `else if` for the chain)

Reply:
Done.